nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -179,13 +179,40 @@ def _get_stat_collector(self, quantizer_config: QuantizerConfig) -> TensorStatis
                                                                         num_samples=self._parameters.number_samples)
         raise RuntimeError('This range type is not supported!')
 
+    def _filter_nncf_graph(self, nncf_graph: NNCFGraph) -> NNCFGraph:
```

Comment:
Why should this method be a part of the `MinMaxQuantization` algorithm?

Reply:
I didn't find better place to put this method. If you have any suggestions - welcome!
So `MinMaxQuantization` should use filtered `NNCFGraph` without Constant subgraphs to avoid incorrect q/dq placements. At the same time "full" `NNCFGraph` could be useful for other algorithms. Also, it has more nice visualization with Constant subgraphs, e.g. weight quantizers.

Reply:
@kshpv, I disagree. `MinMaxQuantization` is the algorithm and should do nothing except that needs to execute its own pipeline. `Filtering NNCFGraph` is not part of any algorithm. It should be placed into one of the model-transformer entities or as a pass of the model preparation for the further quantization pipeline.

Reply:
@kshpv, additionally, if there are any `incorrect q/dq placements` that is the problem of the algorithm that creates the so-called `quantizer_setup`. As I remember, it's the main purpose of the `QuantizerPropagationSolver`, isn't it?

Reply:
@KodiaqQ, yes. I agree with that point. Model-transformer works only on backend-specific model, so it seems there is no good way to put this logic there.

Reply:
@kshpv, it doesn't matter. The quantization algorithm should not care about the model preparations. Since there are not possible to place this functionality into the `ModelTransformer` or somewhere else, it means that `NNCFGraph` builds wrong again.

Reply:
@KodiaqQ there is a good point in general. 
In the signature of our algorithms we pass Model, while, e.g. MinMax doe not utilize it at all, only to build NNCFGraph. Do we need the model in FBC?
I mean maybe we just need to pass NNCFGraph for all algorithms. Is it possible?

Reply:
@kshpv it would be a great next step to use the same general entity like `NNCFGraph` in the algorithms, but currently, it's not possible because `NNCFGraph` did not contain constants and some kind of necessary information. @alexsu52, could you please add details about the usage of the general model representation in the algorithms?

Reply:
@vshampor @alexsu52

Reply:
To my point, some kind of `filtration` logic correlates with the `ignoring pattern` functionality. And it should be implemented as the additional logic near it or as one of the steps for the `NNCFGraph` building.

Reply:
Hi folks, 

Obviously, we have some dual understanding of NNCFGraph. We have to align here and plan improvement steps. I would suggest to schedule offline discussion.

In the context of this PR, the _filter_nncf_grap is common pass of NNCFGraph preparation for `QuantizationPropagationSolver` and I would recomend to call it in the `QuantizationPropagationSolver` implementation.

Reply:
Done